src/main/java/io/lettuce/core/cluster/RedisClusterClient.java

```diff
@@ -556,6 +556,11 @@ <K, V> ConnectionFuture<StatefulRedisConnection<K, V>> connectToNodeAsync(RedisC
         StatefulRedisConnectionImpl<K, V> connection = newStatefulRedisConnection(writer, endpoint, codec,
                 getFirstUri().getTimeout(), getClusterClientOptions().getJsonParser());
 
+        if (RedisAuthenticationHandler.isSupported(getOptions())) {
```

Comment:
again, isnt it a check for the streaming behaviour, why we do it twice at different levels ?

Reply:
See the comment above (https://github.com/redis/lettuce/pull/3068#discussion_r1886864844)
The idea is that you might have a streaming credentials provider but don't want to re-authenticate on some connections.